tools-local/windows/tool-helper.ps1

```diff
@@ -0,0 +1,322 @@
+﻿# Provides helper functions.
+
+function Get-DefaultScriptsFolder
+{
+    [CmdletBinding()]
+    param(
+        [ValidateNotNullOrEmpty()]
+        [parameter(Mandatory=$true, Position=0)]
+        [string]$RepositoryRoot
+    )
+
+    return Join-Path "$RepositoryRoot" "tools-local\windows"
+}
+
+function Get-OperatingSystemArchitecture
+{
+    if ([System.Environment]::Is64BitOperatingSystem)
+    {
+        return "64"
+    }
+
+    return "32"
+}
+
+function Get-ToolVersionsConfig
+{
+    [CmdletBinding()]
+    param(
+        [parameter(Mandatory=$true, Position=0)]
+        [string]$ToolName,
+        [ValidateNotNullOrEmpty()]
+        [parameter(Mandatory=$true, Position=1)]
+        [string]$RepositoryRoot
+    )
+
+    $toolVersionsFilePath = Join-Path "$RepositoryRoot" ".toolversions"
```

Comment:
> What is our plan for official builds with the current approach?

Yes, the plan for official builds is still to use a copy of `.toolversions` that overrides the default one. The path to this file will be passed as a command line argument. I'm thinking we can place that file (not all scripts, just the file only) in a folder whose path can be passed though `OverrideScritpsFolderPath` , and modify `tool-helper` to consume the config values listed in that override file.